### PR TITLE
Include MoveTo Helper class to ign-rendering

### PR DESCRIPTION
**Pull request by ahcorde**
Signed-off-by: ahcorde <ahcorde@gmail.com>

# 🎉 New feature

This PR is part of the consolidation of Scene3D with ign-gazebo's GzScene3D https://github.com/ignitionrobotics/ign-gui/issues/137

## Summary

This class is inside the ign-gazebo plugin which contains a [TODO](https://github.com/ignitionrobotics/ign-gazebo/blob/ign-gazebo5/src/gui/plugins/scene3d/Scene3D.cc#L101-L102) to move this hidden class to ign-rendering.

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [x] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))

**Note to maintainers**: Remember to use **Squash-Merge**

### CODE REVIEW

**Comment by chapulina** [comment]
I like the idea of having this functionality inside `ign-rendering.` I'm not so sure about keeping it in a separate class though. I think it could be part of the camera API, together with all the `Track` and `Follow` functions:

https://github.com/ignitionrobotics/ign-rendering/blob/07c4ca96c924db4004ccab9e0b7a1e6ac5b1b033/include/ignition/rendering/Camera.hh#L196-L273

What do you think, @iche033 ?

**Comment by iche033** [comment]
that was the intention before - putting it somewhere like in BaseCamera class. There is also an advantage of keeping this functionality in a separate class though. We can pimpl'ize it to avoid all the ABI compatibility headaches that we are facing with ign-rendering when making changes to templated classes in header files.

**Comment by chapulina** [comment]
> There is also an advantage of keeping this functionality in a separate class though.

Yeah that's true, the current implementation could be added to `ign-rendering3` even :+1:

**Comment by ahcorde** [comment]
Added pimpl and retargeted to `ign-rendering3`

File: src/MoveToHelper.cc [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+
+  /// \brief Initial pose of the camera used for view angles
+  public: math::Pose3d initCameraPose;
+};
+
+using namespace ignition;
+using namespace rendering;
+
+MoveToHelper::MoveToHelper() :
+  dataPtr(new MoveToHelperPrivate)
```
Comment (mjcarroll) [inline src/MoveToHelper.cc:47]: Nit: `new` isn't needed here.

File: include/ignition/rendering/MoveToHelper.hh [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+#ifndef IGNITION_GUI_MOVETOHELPER_HH_
+#define IGNITION_GUI_MOVETOHELPER_HH_
```
Comment (mjcarroll) [inline include/ignition/rendering/MoveToHelper.hh:18]:
```suggestion
#ifndef IGNITION_RENDERING_MOVETOHELPER_HH_
#define IGNITION_RENDERING_MOVETOHELPER_HH_
```

File: src/MoveToHelper.cc (hunk 1 above)
Comment (ahcorde) [reply src/MoveToHelper.cc:47]: are you sure ? dataPtr is a `unique_ptr`.

Comment (mjcarroll) [reply src/MoveToHelper.cc:48]: Oh yes, sorry.

**Review by iche033 — APPROVED** [review]
looks good to me.

File: src/MoveToHelper.cc (hunk 1 above)
Comment (chapulina) [reply src/MoveToHelper.cc:48]: It's best to use `make_unique`

**Review by chapulina — COMMENTED** [review]
Should we add some tests?

**Comment by ahcorde** [comment]
@iche033 do you mind to have a look to the new tests? https://github.com/ignitionrobotics/ign-rendering/pull/311/commits/8e98a132a8545ad4379e4d4cf70ea79f6d7fa201

File: src/MoveToHelper_TEST.cc [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+  public: void checkIsCompleted();
+
+  public: MoveToHelper moveToHelper;
+
+  public: bool isMoveCompleted = false;
+};
+
+void MoveToHelperTest::OnMoveToComplete()
+{
+  isMoveCompleted = true;
```
Comment (iche033) [inline src/MoveToHelper_TEST.cc:49]: add `this->`

File: src/MoveToHelper_TEST.cc [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+  while (!isMoveCompleted)
+  {
+    moveToHelper.AddTime(0.01);
+    endTime = std::chrono::system_clock::now();
+
+    auto seconds =
+      std::chrono::duration_cast<std::chrono::milliseconds>(
+        endTime - startTime).count() / 1000.0;
+    if (seconds > 0.5)
```
Comment (iche033) [inline src/MoveToHelper_TEST.cc:69]: maybe make max time to wait, i.e 0.5, as arg?

**Review by iche033 — APPROVED** [review]
Just very small nitpicks. This is good to go as is

**Pull request merged**